R/submit.R

```diff
@@ -26,7 +26,7 @@ task_submit <- function(id, ..., resources = NULL,
   }
   root <- hipercow_root(root)
   driver <- hipercow_driver_select(driver, TRUE, root, rlang::current_env())
-  resources <- resources_validate(resources, driver, root)
+  resources <- resources_validate(resources, NULL, driver, root)
```

Comment:
I am not sure about this one in the case where task_submit is called directly - (not via one of the task_create calls) - does this ever happen?

Reply:
I could just name the args driver = driver and root = root, which looks nicer than a naked NULL, but whether task_submit should actually take parallel as an argument...

Reply:
Submit should not need parallel, I think - I think this is related to our nagging concern we have some of the logic of where we save and reference this data from slightly wrong (as discussed with the retry).  For now, `NULL` is fine, name it as an argument though I think.

Another way of doing this that might be nicer is not to do the validation here within resources_validate but do it within the task creation *after* `resources_validate` but before submission?

Reply:
Yeah - I think this is nicer now?